model/pipeline/src/main/proto/external_transforms.proto

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing the external transforms available.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.model.pipeline.v1;
+
+option go_package = "pipeline_v1";
+option java_package = "org.apache.beam.model.pipeline.v1";
+option java_outer_classname = "ExternalTransforms";
+
+import "beam_runner_api.proto";
+import "google/protobuf/duration.proto";
+
+// Payload for Java's GenerateSequence transform
+message GenerateSequencePayload {
```

Comment:
Can this be some sort of a registration mechanism instead of having to modify the proto for every transform that we want to make cross-language compatible ?

Reply:
As long as transforms are bundled with Beam, this should be fine. However, that's a fairly static model and I agree that we want to make them pluggable via some sort of registry.

Reply:
Initially I had something like this in mind. We discussed pro and cons and I implemented this Proto approach here. We will probably support both ways.